apiserver/facades/agent/secretsmanager/secrets.go

```diff
@@ -893,8 +894,13 @@ func (s *SecretsManagerAPI) WatchObsolete(args params.Entities) (params.StringsW
 		return result, errors.Trace(err)
 	}
 	if changes, ok := <-w.Changes(); ok {
-		result.StringsWatcherId = s.resources.Register(w)
-		result.Changes = changes
+		id, err := s.watcherRegistry.Register(w)
+		if err != nil {
+			result.Error = apiservererrors.ServerError(err)
+		} else {
+			result.StringsWatcherId = id
```

Comment:
Shame there isn't uniformity with the `<Thing>WatchResultTypes` (`StringsWatcherId` here versus `WatcherId` elsewhere). Then we might be able to reduce this boilerplate with generics.

Reply:
The whole watcher mess isn't great. Ideally, we should have one watcher per model and emit events on top of the change stream.